starknet/src/proposal_validation_strategies/proposing_power.cairo

```diff
@@ -0,0 +1,85 @@
+#[starknet::contract]
+mod ProposingPowerProposalValidationStrategy {
+    use sx::interfaces::IProposalValidationStrategy;
+    use sx::types::{UserAddress, IndexedStrategy, IndexedStrategyTrait, Strategy};
+    use sx::interfaces::{IVotingStrategyDispatcher, IVotingStrategyDispatcherTrait};
+    use starknet::ContractAddress;
+    use starknet::info;
+    use traits::{Into, TryInto};
+    use option::OptionTrait;
+    use result::ResultTrait;
+    use array::{ArrayTrait, SpanTrait};
+    use serde::Serde;
+    use sx::utils::bits::BitSetter;
+    use box::BoxTrait;
+    use clone::Clone;
+
+    #[storage]
+    struct Storage {}
+
+    #[external(v0)]
+    impl ProposingPowerProposalValidationStrategy of IProposalValidationStrategy<ContractState> {
+        fn validate(
```

Comment:
probably makes sense to follow the same approach as sx-evm where we have the internal `_validate` function defined in `utils` then the strategy can call decode and call whichever ones it wants.

Reply:
0e70874af4f331fea537b7d4ea7f6566d3285da4